src/components/header/components/Nav.tsx

```diff
@@ -1,39 +1,22 @@
-import React, { HTMLProps, useContext } from 'react';
-import classNames from 'classnames';
-import { Container } from '../../layout';
-import { Close as CloseIcon } from '../../icons';
-import HeaderContext, { IHeaderContext } from '../HeaderContext';
+import React, { HTMLProps } from 'react';
+import NavTitle from './NavTitle';
+import NavItemList from './NavItemList';
+import NavMenuCollapse from './NavMenuCollapse';
+import NavContainer from './NavContainer';
 
 const Nav: React.FC<HTMLProps<HTMLDivElement>> = ({
   className,
   children,
-  onClick,
   open,
   ...rest
-}) => {
-  const { menuOpen, toggleMenu } = useContext<IHeaderContext>(HeaderContext);
-
-  return (
-    <nav
-      className={classNames(
-        'nhsuk-header__navigation',
-        { 'js-show': open !== undefined ? open : menuOpen },
-        className,
-      )}
-      {...rest}
-    >
-      <Container>
-        <p className="nhsuk-header__navigation-title">
-          <span>Menu</span>
-          <button className="nhsuk-header__navigation-close" type="button" onClick={toggleMenu}>
-            <CloseIcon />
-            <span className="nhsuk-u-visually-hidden">Close menu</span>
-          </button>
-        </p>
-        <ul className="nhsuk-header__navigation-list">{children}</ul>
-      </Container>
-    </nav>
-  );
-};
+}) => (
+  <NavContainer open={open} {...rest}>
```

Comment:
TODO: This needs to have `className` passed down to keep the existing interface after refactor.